hedera-node/docs/design/services/smart-contract-service/RedirectContract-description.md

```diff
@@ -0,0 +1,263 @@
+     prohibits a _delegate_ call to a system contract method:  If it is done by the system by using
+     its redirect contract.  (See `FrameUtils.callTypeOf()`.)
+
+## Other decompiles/disassemblies of the redirect contract
+
+### 3-address
+
+```
+function __function_selector__() public {
+    Begin block 0x0
+    prev=[], succ=[0xb, 0xf]
+    =================================
+    0x0: v0(0x80) = CONST
+    0x2: v2(0x40) = CONST
+    0x4: MSTORE v2(0x40), v0(0x80)
+    0x5: v5 = CALLVALUE
+    0x7: v7 = ISZERO v5
+    0x8: v8(0xf) = CONST
+    0xa: JUMPI v8(0xf), v7
+
+    Begin block 0xb
+    prev=[0x0], succ=[]
+    =================================
+    0xb: vb(0x0) = CONST
+    0xe: REVERT vb(0x0), vb(0x0)
+
+    Begin block 0xf
+    prev=[0x0], succ=[0x58, 0x54]
+    =================================
+    0x11: v11(0x0) = CONST
+    0x13: v13(0x167) = CONST
+    0x18: v18(0x618dc65efefefefefefefefefefefefefefefefefefefefe) = CONST
+    0x31: v31(0x0) = CONST
+    0x33: MSTORE v31(0x0), v18(0x618dc65efefefefefefefefefefefefefefefefefefefefe)
+    0x34: v34 = CALLDATASIZE
+    0x35: v35(0x0) = CONST
+    0x37: v37(0x20) = CONST
+    0x39: CALLDATACOPY v37(0x20), v35(0x0), v34
+    0x3a: v3a(0x0) = CONST
+    0x3d: v3d = CALLDATASIZE
+    0x3e: v3e(0x18) = CONST
+    0x40: v40 = ADD v3e(0x18), v3d
+    0x41: v41(0x8) = CONST
+    0x44: v44 = GAS
+    0x45: v45 = DELEGATECALL v44, v13(0x167), v41(0x8), v40, v3a(0x0), v3a(0x0)
+    0x46: v46 = RETURNDATASIZE
+    0x48: v48(0x0) = CONST
+    0x4b: RETURNDATACOPY v48(0x0), v48(0x0), v46
+    0x4d: v4d(0x0) = CONST
+    0x50: v50 = EQ v45, v4d(0x0)
+    0x51: v51(0x58) = CONST
+    0x53: JUMPI v51(0x58), v50
+
+    Begin block 0x58
+    prev=[0xf], succ=[]
+    =================================
+    0x5a: v5a(0x0) = CONST
+    0x5c: REVERT v5a(0x0), v46
+
+    Begin block 0x54
+    prev=[0xf], succ=[]
+    =================================
+    0x55: v55(0x0) = CONST
+    0x57: RETURN v55(0x0), v46
+
+}
+```
+
+### Disassembly
+
+Via `library.dedaub.com/decompile`:
+
+```
+ 0x0: PUSH1     0x80
+ 0x2: PUSH1     0x40
+ 0x4: MSTORE
+ 0x5: CALLVALUE
+ 0x6: DUP1
+ 0x7: ISZERO
+ 0x8: PUSH1     0xf
+ 0xa: JUMPI
+ 0xb: PUSH1     0x0
+ 0xd: DUP1
+ 0xe: REVERT
+ 0xf: JUMPDEST
+0x10: POP
+0x11: PUSH1     0x0
+0x13: PUSH2     0x167
+0x16: SWAP1
+0x17: POP
+0x18: PUSH24    0x618dc65efefefefefefefefefefefefefefefefefefefefe
+0x31: PUSH1     0x0
+0x33: MSTORE
+0x34: CALLDATASIZE
+0x35: PUSH1     0x0
+0x37: PUSH1     0x20
+0x39: CALLDATACOPY
+0x3a: PUSH1     0x0
+0x3c: DUP1
+0x3d: CALLDATASIZE
+0x3e: PUSH1     0x18
+0x40: ADD
+0x41: PUSH1     0x8
+0x43: DUP5
+0x44: GAS
+0x45: DELEGATECALL
+0x46: RETURNDATASIZE
+0x47: DUP1
+0x48: PUSH1     0x0
+0x4a: DUP1
+0x4b: RETURNDATACOPY
+0x4c: DUP2
+0x4d: PUSH1     0x0
+0x4f: DUP2
+0x50: EQ
+0x51: PUSH1     0x58
+0x53: JUMPI
+0x54: DUP2
+0x55: PUSH1     0x0
+0x57: RETURN
+0x58: JUMPDEST
+0x59: DUP2
+0x5a: PUSH1     0x0
+0x5c: REVERT
+0x5d: INVALID
+0x5e: LOG2
```

Comment:
it looks like it's disassembling the metadata, should we mention this to avoid confusion to the reader?

Reply:
I'll replace the bad disassembly of the metadata w/ a comment saying "remaining bytes are metadata".

(Wonder why they don't fix this?  My disassembler - no longer in the repo at `main` - figured out where the metadata was and dumped it differently ...)